src/modules/numberToWords/index.ts

```diff
@@ -1,132 +1,69 @@
 
-const config = {
-	scale: ["", "هزار", "میلیون", "میلیارد"],
-};
+const numberToWords: NumberToWordsType = (inputNumber, options) => {
+	const isNumberValid = (n: number) => typeof n === "number" && Number.isSafeInteger(n) && n !== 0;
+	const isNegative = (n: number) => n < 0;
+	const notValidNumberError = () => TypeError("the number must be a safe integer value");
 
-const numberToWord: INumberToWord = {};
+	if (typeof inputNumber !== "string" && !Number.isSafeInteger(inputNumber)) return notValidNumberError();
+	const number = Number(typeof inputNumber === "number" ? inputNumber : removeCommas(inputNumber));
+	const isOrdinal = options?.ordinal || false;
 
-numberToWord[0] = "";
-numberToWord[1] = "یک";
-numberToWord[2] = "دو";
-numberToWord[3] = "سه";
-numberToWord[4] = "چهار";
-numberToWord[5] = "پنج";
-numberToWord[6] = "شش";
-numberToWord[7] = "هفت";
-numberToWord[8] = "هشت";
-numberToWord[9] = "نه";
-numberToWord[10] = "ده";
-numberToWord[11] = "یازده";
-numberToWord[12] = "دوازده";
-numberToWord[13] = "سیزده";
-numberToWord[14] = "چهارده";
-numberToWord[15] = "پانزده";
-numberToWord[16] = "شانزده";
-numberToWord[17] = "هفده";
-numberToWord[18] = "هجده";
-numberToWord[19] = "نوزده";
-numberToWord[20] = "بیست";
-numberToWord[30] = "سی";
-numberToWord[40] = "چهل";
-numberToWord[50] = "پنجاه";
-numberToWord[60] = "شصت";
-numberToWord[70] = "هفتاد";
-numberToWord[80] = "هشتاد";
-numberToWord[90] = "نود";
-numberToWord[100] = "صد";
-numberToWord[200] = "دویست";
-numberToWord[300] = "سیصد";
-numberToWord[400] = "چهار صد";
-numberToWord[500] = "پانصد";
-numberToWord[600] = "شش صد";
-numberToWord[700] = "هفت صد";
-numberToWord[800] = "هشت صد";
-numberToWord[900] = "نه صد";
+	const getWord = (n: number) => numbersWordList.get(n) ?? "";
+	const addNegativeSuffix = (str: string) => "منفی" + " " + str;
 
-const toWords = (number: number): string => {
-	let unit = 100;
-	let result = "";
+	function transformeToWord(num: number): string {
+		if (num === 0) return "";
+		if (num <= 9) return getWord(num);
+		else if (num >= 11 && num <= 19) return getWord(num);
 
-	while (unit > 0) {
-		if (Math.floor(number / unit) * unit !== 0) {
-			if (number in numberToWord) {
-				result += numberToWord[number];
-				break;
-			} else {
-				result += numberToWord[Math.floor(number / unit) * unit] + " و ";
-				number %= unit;
-			}
-		}
-		unit = Math.floor(unit / 10);
+		const residual = num <= 99 ? num % 10 : num % 100;
+		return residual === 0 ? getWord(num) : `${getWord(num - residual)} و ${transformeToWord(residual)}`;
 	}
-	return result;
-};
-
-/**
- *
- * Convert to words
- *
- * @category Number conversion
- * @return Converted numbers to words. e.g: صد و بیست و یک
- */
-function numberToWords(number: bigint | number | string, { ordinal = false }: NumberToWordsOptions = {}): string {
-	if (typeof number === "undefined") return "";
 
-	if (number === 0) {
-		return "صفر";
-	}
+	function performer(num: number) {
```

Comment:
What is this function? what does it do? Write document above the function and also add return type.